src/components/main/Main.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+
+import { ContentItem } from '../contentItem/ContentItem';
+import { usePizzasStore } from '../../store/usePizzasStore';
+
+import { PizzaType } from '../../ts/types/types';
+
+import './Main.scss';
+
+export const Main = () => {
+  const getPizzas = usePizzasStore((state: any) => state.getPizzas); //any
+  const pizzas = usePizzasStore((state: any) => state.data); //any
+  const isLoading = usePizzasStore((state: any) => state.isLoading); //any
+
+  useEffect(() => {
+    getPizzas();
+  }, []);
```

Comment:
Looks like the Frankenstein's monster. Pizzas are basically immutable, so there is no need to keep them in the state at all. You can just load them once on a startup using plain JS.

If you want your data fetching to be more realistic, check out the caching approach: `React Query` or `SWR`. If you want to avoid waterfalls as well, check out prefetching on route level (see React Router v6).